src/enumerable.cr

```diff
@@ -238,7 +238,7 @@ module Enumerable(T)
   # ```
   # [1, 2, 3, 4].count { |i| i % 2 == 0 } # => 2
   # ```
-  def count
+  def count(& : T -> _) : Int32
```

Comment:
The non-yielding overload (right below) received a `Int32` return type restriction in #10582. But this restriction only holds if the yielding overload returns `Int32` as well (which is in fact the case). So I figured we should just add it for consistency.